samples/web/content/apprtc/apprtc.py

```diff
@@ -423,32 +478,17 @@ def send_message_to_collider(self, room_id, client_id, message):
 
   def post(self, room_id, client_id):
     message_json = self.request.body
-    with LOCK:
-      client_map = get_room_client_map(room_id)
-      occupancy = len(client_map)
-      # Check that room exists.
-      if occupancy == 0:
-        logging.warning('Unknown room: ' + room_id)
-        self.write_response('UNKNOWN_ROOM')
-        return
-
-      # Check that client is registered.
-      if not client_id in client_map:
-        logging.warning('Unknown client: ' + client_id)
-        self.write_response('UNKNOWN_CLIENT')
-        return
+    result = save_message_from_client(
+        self.request.host_url, room_id, client_id, message_json)
+    if result['error'] is not None:
+      self.write_response(result['error'])
+      return
+    self.write_response('SUCCESS')
+    if result['saved']:
```

Comment:
We do, because 'saved' could be false even if 'error' is None, when the
second user already joined.

On Fri, Jan 9, 2015 at 1:23 PM, Justin Uberti <notifications@github.com>
wrote:

> In samples/web/content/apprtc/apprtc.py
> <https://github.com/GoogleChrome/webrtc/pull/330#discussion-diff-22743972>
> :
>
> > -        logging.warning('Unknown room: ' + room_id)
> > -        self.write_response('UNKNOWN_ROOM')
> > -        return
> > -
> > -      # Check that client is registered.
> > -      if not client_id in client_map:
> > -        logging.warning('Unknown client: ' + client_id)
> > -        self.write_response('UNKNOWN_CLIENT')
> > -        return
> > +    result = save_message_from_client(
> > +        self.request.host_url, room_id, client_id, message_json)
> > +    if result['error'] is not None:
> > +      self.write_response(result['error'])
> > +      return
> > +    self.write_response('SUCCESS')
> > +    if result['saved']:
>
> So do we need 'saved' then?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/webrtc/pull/330/files#r22743972>.
>



-- 
Jiayang Liu